### PR TITLE
feat(webhooks): enhance webhook subscription handling

### PR DESCRIPTION
**Pull request by alonp99**
- Update SubscriptionSchema to support 'email' alongside 'webhook'
- Implement mergeSubscriptions function to handle subscription merging
- Introduce unit tests for mergeSubscriptions functionality

(your webhook subscriptions are now better organized than your last family reunion)

## Release Notes

- **New Features**
	- Enhanced subscription handling by supporting both webhook and email subscription types.
	- Improved webhook processing to incorporate customer-specific subscription data.
	- Introduced a new function for merging customer and workflow subscriptions.

- **Bug Fixes**
	- Updated webhook retrieval logic to handle more flexible subscription scenarios.

- **Refactor**
	- Optimized subscription merging mechanism for more dynamic event handling.
	- Expanded customer data retrieval to include subscription information.
	- Enhanced unit tests for the rule engine to cover a wider range of scenarios and edge cases.

### CODE REVIEW

File: services/workflows-service/src/events/get-webhooks.ts [hunk 1]
```diff
@@ -12,12 +13,53 @@ export type Webhook = {
   };
 };
 
+export const mergeSubscriptions = (
+  customerSubscriptions: TCustomerSubscription['subscriptions'],
+  workflowSubscriptions: TCustomerSubscription['subscriptions'],
+): TCustomerSubscription['subscriptions'] => {
+  if (!workflowSubscriptions?.length) return customerSubscriptions ?? [];
```
Comment (rzssh) [inline services/workflows-service/src/events/get-webhooks.ts:20]: we could add an eslint rule to enforce either always putting curly braces after if/else blocks or allow
such notation, but make it consistent across the codebase.

**Review by chesterkmr — APPROVED** [review]

**Review by r4zendev — APPROVED** [review]
LGTM

File: services/workflows-service/src/events/get-webhooks.ts (hunk 1 above)
Comment (tomer-shvadron) [reply services/workflows-service/src/events/get-webhooks.ts:20]: I agree with Sasha, personally I hate `if (...) return`

File: services/workflows-service/src/events/workflow-completed-webhook-caller.ts [hunk 2]
```diff
@@ -49,18 +49,20 @@ export class WorkflowCompletedWebhookCaller {
...
-    );
-
     const customer = await this.customerService.getByProjectId(data.runtimeData.projectId, {
       select: {
         authenticationConfiguration: true,
+        subscriptions: true,
       },
     });
 
+    const webhooks = getWebhooks(
```
Comment (tomer-shvadron) [inline services/workflows-service/src/events/workflow-completed-webhook-caller.ts]: Can we change the `getWebhooks` function to receive names parameters?
Something like:
```suggestion
    const webhooks = getWebhooks({
      config: ...,
      subscriptions: ...
      envName: ...
      eventName: ...
    });
```

**Review by tomer-shvadron — APPROVED** [review]

File: services/workflows-service/src/events/get-webhooks.ts (hunk 1 above)
Comment (alonp99) [reply services/workflows-service/src/events/get-webhooks.ts:20]: Adding tech debt task to add it to eslint base config
